TELCODOCS-2329 - ACM PolicyGenerator RAN limitations (#93606)
openshift/openshift-docs: modules/cnf-topology-aware-lifecycle-manager-troubleshooting.adoc

----
$ oc describe job -n openshift-talo-pre-cache pre-cache
----

[discrete]

> [id="talo-troubleshooting-pre-placement-tolerations_{context}"]

Reviewer: i know it has nothing to do with the PR, but can the id talo be replaced by talm everywhere?

=== Matching policies and `ManagedCluster` CRs before the `ManagedCluster` CRs are available

Issue:: You want {rh-rhacm} to match policies and `ManagedCluster` CRs before the `ManagedCluster` CRs are available.

> Resolution:: By default, {rh-rhacm} does not match policies and `ManagedCluster` CRs until the `ManagedCluster` CRs are available. The {rh-rhacm} policy generator automatically generates the `Placement` CR which includes the tolerations needed to enable the match before the `ManagedCluster` CRs are available. If you use a custom `Placement` CR instead, you should include the tolerations in that `Placement` CR.

Reviewer 2: ```suggestion
Resolution:: By default, {rh-rhacm} does not match policies and `ManagedCluster` CRs until the `ManagedCluster` CRs are available. The {rh-rhacm} `PolicyGenerator` automatically generates the `Placement` CR which includes the tolerations needed to enable the match before the `ManagedCluster` CRs are available. If you use a custom `Placement` CR instead, you should include the tolerations in that `Placement` CR.
```

> Resolution:: By default, {rh-rhacm} does not match policies and `ManagedCluster` CRs until the `ManagedCluster` CRs are available. The {rh-rhacm} `{policy-gen-cr}` automatically generates the `Placement` CR which includes the tolerations needed to enable the match before the `ManagedCluster` CRs are available. If you use a custom `Placement` CR instead, you should include the tolerations in that `Placement` CR.

Author: Thanks @sabbir-47 
we've reworked the text and added context

----
$ oc describe job -n openshift-talo-pre-cache pre-cache
----

[discrete]
[id="talo-troubleshooting-pre-placement-tolerations_{context}"]
=== Matching policies and `ManagedCluster` CRs before the managed cluster is available

Issue:: You want {rh-rhacm} to match policies and managed clusters before the managed clusters become available.
+

> Resolution:: To ensure that {cgu-operator} correctly applies the {rh-rhacm} policies specified in the `spec.managedPolicies` field of the `ClusterGroupUpgrade` (CGU) CR, {cgu-operator} needs to match these policies to the managed cluster before the managed cluster is available. 

Reviewer 3: - Expand CR for its first occurrence

> The {rh-rhacm} `PolicyGenerator` uses the generated `Placement` CR to do this automatically. 
> By default, this `Placement` CR includes the necessary tolerations. 
> This ensures proper {cgu-operator} behavior. 

Reviewer 3: ```suggestion
The {rh-rhacm} `PolicyGenerator` uses the generated `Placement` CR to do this automatically. 
By default, this `Placement` CR includes the necessary tolerations to ensure proper {cgu-operator} behavior. 
```

+

> The expected `spec.tolerations` settings in the `Placement` CR are:

Reviewer 4: ```suggestion
The expected `spec.tolerations` settings in the `Placement` CR are as follows:
```
From [ISG](https://www.ibm.com/docs/en/ibm-style?topic=punctuation-colons#introductory-text): To avoid translation problems, make introductions complete sentences.

Author: thanks for the revewi @lahinson

+

> [source,terminal]

Reviewer 3: ```suggestion
[source,yaml]
```

----
  tolerations:
    - key: cluster.open-cluster-management.io/unavailable
     operator: Exists
    - key: cluster.open-cluster-management.io/unreachable

>      operator: Exists

Reviewer 3: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#signs-and-symbols-in-code-blocks

> For snippets or sections of a file, use an ellipsis (…​) to show that the file continues before or after the quoted block. For YAML, use ('#…​') instead.

----
+

> If you use a custom `Placement` CR instead of the one generated by the {rh-rhacm} `PolicyGenerator`, you should include these tolerations in that `Placement` CR.

Reviewer 4: ```suggestion
If you use a custom `Placement` CR instead of the one generated by the {rh-rhacm} `PolicyGenerator`, include these tolerations in that `Placement` CR.
```
ISG Word usage guidance about "should": https://www.ibm.com/docs/en/ibm-style?topic=word-usage#should
